github-download-release/action.yml

```diff
@@ -0,0 +1,62 @@
+name: Github Download Release
+description: Downloads the specified release using gh client
+inputs:
+  token:
+    description: The authorization token for the target repo
+    required: true
+  github-version:
+    description: Github version
+    required: false
+    default: 'v2.0.0'
+  owner:
+    description: The repo owner
+    required: true
+  repo:
+    description: The repo name
+    required: true
+  release:
+    description: The release tag or number
+    required: true
+  format-ext:
+    description: The type of archive
+    required: false
+    default: linux_amd64.tar.gz
+outputs:
+  filename:
+    description: The name of the downloaded tarball
+    value: ${{ steps.download-tarball.outputs.filename }}
+runs:
+  using: composite
+  steps:
+    - name: Download github cli
```

Comment:
is there a reason to install the GH CLI at runtime? github-hosted runners should have it installed already

Reply:
Yeah a github hosted runner would be idea. The repo being downloaded is this terraform-pipeline-executor project written in go. What it's doing under the hood requires the authentications in our self hosted runner.

Reply:
ah ok, so this is intended to be ran on self-hosted runners. A pattern I've seen for these types of actions is to check if the tool is already present, and only install if it's not present. If this action is only for use on self-hosted runners then maybe that's overkill, but it would be helpful to make it very obvious that the action shouldn't be used on gh-hosted runners. (maybe that's true of this entire repo and I just missed it?)

Reply:
Yeah they're pretty much all meant to be used in a situation where you have to use a self-hosted a runner. The discover-changed action is the only one that's used in a gh hosted runner. I could check for prior existence, however to be prudent, I would also need to check if the version of an already-present binary matches, as the user might want to override the version already installed. That said, for our specific case, if I had it my way, I would have control of the runner image and make sure the gh cli was on there to begin with. I think I'm landing on explaining the intent of use in the README

Reply:
I could also add a `skip-gh-if-already-present` boolean option, whereby if it's present, and the user does not care about the version, then to skip the download of gh.

Reply:
I ended up doing away with gc cli altogether, and figured out the correct way to do a curl request on private repo.